### PR TITLE
feat: add rule `sort-keys` (fixes #104)

### LINKED ISSUE #104: Add rule to enforce sorting of the properties
Equivalent to http://eslint.org/docs/rules/sort-keys

### PR DESCRIPTION
**Pull request by aaron-harvey**

### CODE REVIEW

File: .README/rules/sort-keys.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+The first option specifies sort order.
+
+* `"asc"` (default) - enforce ascending sort order.
+* `"desc"` - enforce descending sort order.
+
+The second option takes an object with two possible properties.
+
+* `caseSensitive` - if `true`, enforce case-sensitive sort order. Default is `true`.
+* `natural` - if `true`, enforce [natural sort order](https://en.wikipedia.org/wiki/Natural_sort_order). Default is `false`.
```
Comment (gajus) [inline .README/rules/sort-keys.md]: Thanks for including the wikipedia link. The natural order was not obvious to me when I first explored ESLint sort-keys rule.

File: .README/rules/sort-keys.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+```js
+{
+  "rules": {
+    "flowtype/sort-keys": [ 2, "asc", {
```
Comment (gajus) [inline .README/rules/sort-keys.md]: Please use one property per line, like so:

```json
{
  "rules": {
    "flowtype/sort-keys": [
    	2,
    	"asc",
    	{
        "caseSensitive": true,
        "natural": false
      }
    ]
  }
}
```

Comment (aaron-harvey) [inline .README/rules/sort-keys.md]: I updated the documentation formatting in the most recent commit

**Comment by danharper** [comment]
👍 Thanks for this. Been busy the past few days, I'll try this out tomorrow :)

**Comment by aaron-harvey** [comment]
I rebased this branch on v2.18.2 - the check failure looks like a `valid-syntax` issue.

File: src/index.js [hunk 3]
```diff
@@ -15,7 +15,8 @@ import validSyntax from './rules/validSyntax';
 import booleanStyle from './rules/booleanStyle';
 import delimiterDangle from './rules/delimiterDangle';
 import noDupeKeys from './rules/noDupeKeys';
-import recommended from './configs/recommended.json';
+import sortKeys from './rules/sortKeys';
+import recommended from './configs/recommended';
```
Comment (danharper) [inline src/index.js]: ^ I think this got messed up in the rebase. It should still be

```js
import recommended from './configs/recommended.json';
```

**Comment by danharper** [comment]
Merged as https://github.com/gajus/eslint-plugin-flowtype/commit/f6b8deb187decfd0dc91262961444052c848a6a1. Thanks :)

**Comment by danharper** [comment]
Shipped as 2.19.0

File: src/index.js (hunk 3 above)
Comment (aaron-harvey) [inline src/index.js]: ah missed that!

**Pull request closed without merging**